File: src/main/java/com/polygloat/exceptions/AuthenticationException.java
package com.polygloat.exceptions;

import com.polygloat.constants.Message;
import org.springframework.http.HttpStatus;

import java.io.Serializable;
import java.util.List;

public class AuthenticationException extends ErrorException {
    public AuthenticationException(Message message, List<Serializable> params) {
        super(message, params);
    }

    public AuthenticationException(Message message) {
        super(message);
    }

    public HttpStatus getHttpStatus() {
        return HttpStatus.UNAUTHORIZED;
    }
}

File: backend/data/src/main/kotlin/io/tolgee/exceptions/AuthenticationException.kt
package io.tolgee.exceptions

import io.tolgee.constants.Message
import org.springframework.http.HttpStatus
import java.io.Serializable

open class AuthenticationException : ErrorException {
  constructor(message: Message) : super(message)

  constructor(message: Message, params: List<Serializable?>? = null, cause: Exception? = null) : super(
    message,
    params,
    cause,
  )

  constructor(code: String, params: List<Serializable?>? = null, cause: Exception? = null) : super(code, params, cause)

  override val httpStatus: HttpStatus
    get() = HttpStatus.UNAUTHORIZED
}
